com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/archive/ResourceDigestCache.java

```diff
@@ -0,0 +1,117 @@
+// Copyright 2020-2023 The Defold Foundation
+// Copyright 2014-2020 King
+// Copyright 2009-2014 Ragnar Svensson, Christian Murray
+// Licensed under the Defold License version 1.0 (the "License"); you may not use
+// this file except in compliance with the License.
+// 
+// You may obtain a copy of the License, together with FAQs at
+// https://www.defold.com/license
+// 
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+package com.dynamo.bob.archive;
+
+import java.util.Map;
+import java.util.HashMap;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+import com.dynamo.bob.fs.IResource;
+import com.dynamo.bob.CompileExceptionError;
+import com.dynamo.bob.archive.ManifestBuilder;
+
+public class ResourceDigestCache {
```

Comment:
Cache for resource digests. Primary purpose is to cache the digest (hash) from the LuaBuilder until we build the archive.